react/utils/cookie.ts

```diff
@@ -43,7 +45,7 @@ export function getFacetsData(facetsDataTarget: string) {
 
   //  In case the facets came from the shipping_info cookie we must replace ":" by ";" because ";" is not allowed in cookies.
   const facetsTarget = facets
-    .replace(':', ';')
+    .replace(ALL_COLON_REGEX, ';')
```

Comment:
⛏️ I believe this facets/cookies handling could be better encapsulated together. Perhaps you could encode this cookie value into base64 and decode it while retrieving it: 
<img width="268" alt="Screenshot 2025-01-17 at 09 35 56" src="https://github.com/user-attachments/assets/143749ca-6f54-491b-b9ca-5c0be5cfa6e9" />

Reply:
Thanks, it's a good idea! I'll do this change to add the encode to base64. 🚀